source/common/upstream/cluster_factory_impl.cc

```diff
@@ -96,8 +96,22 @@ ClusterFactoryImplBase::selectDnsResolver(const envoy::config::cluster::v3::Clus
     for (const auto& resolver_addr : resolver_addrs) {
       resolvers.push_back(Network::Address::resolveProtoAddress(resolver_addr));
     }
-    const bool use_tcp_for_dns_lookups = cluster.use_tcp_for_dns_lookups();
-    return context.dispatcher().createDnsResolver(resolvers, use_tcp_for_dns_lookups);
+    envoy::config::core::v3::DnsResolverOptions dns_resolver_options;
+    if (cluster.has_dns_resolver_options()) {
+      const auto& cluster_dns_options = cluster.dns_resolver_options();
+      dns_resolver_options.set_use_tcp_for_dns_lookups(
+          cluster_dns_options.use_tcp_for_dns_lookups());
+      dns_resolver_options.set_no_default_search_domain(
+          cluster_dns_options.no_default_search_domain());
```

Comment:
Can you use something like `CopyFrom` here?
Otherwise, more fields will require manual copying.

Reply:
done. thanks.